packages/main/cypress/specs/TimePicker.cy.tsx

```diff
@@ -0,0 +1,390 @@
+import "../../src/Assets.js";
+import { setLanguage } from "@ui5/webcomponents-base/dist/config/Language.js";
+import TimePicker from "../../src/TimePicker.js";
+import Label from "../../src/Label.js";
+
+describe("TimePicker Tests", () => {
+	it("input receives value in format pattern depending on the set language", () => {
+		cy.wrap({ setLanguage })
+			.invoke("setLanguage", "bg");
+
+		cy.mount(<TimePicker value="03:16:16"></TimePicker>);
+
+		cy.get("[ui5-time-picker]")
+			.as("timePicker");
+
+		cy.get<TimePicker>("@timePicker")
+			.ui5TimePickerGetInnerInput()
+			.should("have.value", "3:16:16 ч.");
+
+		cy.wrap({ setLanguage })
+			.invoke("setLanguage", "en");
+	});
+
+	it("tests clocks value", () => {
+		cy.mount(<TimePicker formatPattern="HH:mm:ss" value="11:12:13"></TimePicker>);
+
+		cy.get("[ui5-time-picker]")
+			.as("timePicker")
+			.ui5TimePickerValueHelpIconPress();
+
+		cy.get<TimePicker>("@timePicker")
+			.ui5TimePickerGetClock("hours")
+			.should("have.prop", "selectedValue", 11);
+
+		cy.get<TimePicker>("@timePicker")
+			.ui5TimePickerGetClock("minutes")
+			.should("have.prop", "selectedValue", 12);
+
+		cy.get<TimePicker>("@timePicker")
+			.ui5TimePickerGetClock("seconds")
+			.should("have.prop", "selectedValue", 13);
+	});
+
+	it("tests clocks submit value", () => {
+		cy.mount(<TimePicker formatPattern="hh:mm:ss" value="12:00:01"></TimePicker>);
+
+		function pressKeyNTimes(key: "ArrowDown" | "ArrowUp" | "Space" | "Tab" | "Enter", n: number) {
+			for (let i = 0; i < n; i++) {
+				cy.realPress(key);
+			}
+		}
+
+		cy.get("[ui5-time-picker]")
+			.as("timePicker")
+			.ui5TimePickerValueHelpIconPress();
+
+		cy.get("@timePicker")
+			.ui5TimePickerGetClock("hours")
+			.realClick();
+
+		pressKeyNTimes("ArrowDown", 10);
+		cy.realPress("Space");
+
+		cy.get("@timePicker")
+			.ui5TimePickerGetClock("minutes");
+
+		pressKeyNTimes("ArrowDown", 22);
+		pressKeyNTimes("ArrowUp", 2);
+
+		cy.realPress("Space");
+
+		cy.get("@timePicker")
+			.ui5TimePickerGetClock("seconds");
+
+		pressKeyNTimes("ArrowUp", 4);
+
+		cy.realPress("Tab");
+		cy.realPress("Enter");
+
+		cy.get("@timePicker")
+			.should("have.value", "02:40:05");
+	});
+
+	it("tests submit wrong value", () => {
+		cy.mount(<TimePicker></TimePicker>);
+
+		cy.get("[ui5-time-picker]")
+			.as("timePicker")
+			.ui5TimePickerGetInnerInput()
+			.realClick()
+			.should("be.focused")
+			.realType("123123123")
```

Comment:
realType and realPress should not be chained with previous element. split to cy.realType and cy.realPress